base/math.jl

```diff
@@ -698,7 +698,9 @@ end
 @inline ^(x::Float64, y::Integer) = x ^ Float64(y)
 @inline ^(x::Float32, y::Integer) = x ^ Float32(y)
 @inline ^(x::Float16, y::Integer) = Float16(Float32(x) ^ Float32(y))
-@inline ^{p}(x::Float16, ::Type{Val{p}}) = Float16(Float32(x) ^ Val{p})
+@inline ^{p}(x::Float16, ::Type{Val{p}}) = Float16(x ^ p)
+@inline ^{p}(x::Float32, ::Type{Val{p}}) = Float32(x ^ p)
+@inline ^{p}(x::Float64, ::Type{Val{p}}) = Float64(x ^ p)
 
```

Comment:
@tkelman, yes, I commented on those below — those methods can now just be deleted.  (The fallback `literal_pow` will now do the right thing for those types.)